### PR TITLE
refunder: expand E2E tests

### PR DESCRIPTION
**Pull request by tilacog**
Resolves: https://github.com/cowprotocol/services/issues/3785

### Description

This PR improves test coverage for the refunder crate by expanding the end-to-end tests covering many refund scenarios.

#### Other Changes
1. Promoted the `RefundStatus` from tests to the main crate logic. 
1. Fixed compilation error in `shared` crate that was blocking this work.

Run the refunder E2E tests:
```
cargo nextest run -p e2e refunder
```

### CODE REVIEW

File: crates/refunder/src/refund_service.rs [hunk 1]
```diff
@@ -255,34 +183,29 @@ impl RefundService {
...
-                        .inspect_err(|err| tracing::error!(?err, "failed to get data from db"))
-                        .ok()
-                })
+                .filter_map(|result| async { result.ok() })
                 .collect()
                 .await;
             self.submitter
-                .submit(uids, encoded_ethflow_orders, contract)
+                .submit_refund(&uids, encoded_ethflow_orders, contract)
                 .await?;
```
[the remark this answers is not in the document]
Comment (tilacog) [reply crates/refunder/src/refund_service.rs:302]: I think this is the same problem that’s been reported here:

1. https://github.com/cowprotocol/services/blob/f79f7d109cb6a293fd91606739713680435cf938/crates/refunder/src/refund_service.rs#L523-L539
1. https://github.com/cowprotocol/services/blob/f79f7d109cb6a293fd91606739713680435cf938/crates/refunder/src/refund_service.rs#L585-L608



I wasn't aiming for any behavioral changes with this PR, but happy to review and discuss this during the review.

File: crates/shared/src/lib.rs
Comment (jmg-duarte) [inline crates/shared/src/lib.rs:1]: Why the change? This pertains to the address::random, I failed to quote the line, my bad

File: crates/refunder/src/traits.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+use {
+    alloy::primitives::{Address, B256, address},
+    anyhow::Result,
+    contracts::alloy::CoWSwapEthFlow::{self, EthFlowOrder},
+    database::{OrderUid, ethflow_orders::EthOrderPlacement},
+};
+
+const NO_OWNER: Address = Address::ZERO;
+const INVALIDATED_OWNER: Address = address!("0xffffffffffffffffffffffffffffffffffffffff");
```
Comment (jmg-duarte) [inline crates/refunder/src/traits.rs]:
```suggestion
const INVALIDATED_OWNER: Address = Address::repeat_byte(0xff);
```

File: crates/refunder/src/refund_service.rs [hunk 3]
```diff
@@ -1,252 +1,180 @@
...
+        );
+    }
+    to_be_refunded_uids
+}
+
+pub struct RefundService<D, C, S>
+where
+    D: DbRead,
+    C: ChainRead,
+    S: ChainWrite,
```
Comment (jmg-duarte) [inline crates/refunder/src/refund_service.rs]:
```suggestion
    D: DbRead,
    CR: ChainRead,
    CW: ChainWrite,
```

File: crates/refunder/src/refund_service.rs [hunk 4]
```diff
@@ -1,252 +1,180 @@
...
+    pub submitter: S,
+    pub min_validity_duration: i64,
+    pub min_price_deviation: f64,
+}
+
+impl<D, C, S> RefundService<D, C, S>
+where
+    D: DbRead,
+    C: ChainRead,
+    S: ChainWrite,
```
Comment (jmg-duarte) [inline crates/refunder/src/refund_service.rs]: Ditto

File: crates/refunder/src/refund_service.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmg-duarte) [reply crates/refunder/src/refund_service.rs:302]: IMO we should fix in a separate PR

File: crates/refunder/src/refund_service.rs [hunk 5]
```diff
@@ -292,59 +215,696 @@ impl RefundService {
...
-                max_gas_price: 0,
-                start_priority_fee_tip: 0,
-            },
+    /// Test addresses with semantic meaning for filtering logic.
+    pub const KNOWN_ETHFLOW: Address = address!("0x2222222222222222222222222222222222222222");
+    pub const KNOWN_ETHFLOW_2: Address = address!("0x4444444444444444444444444444444444444444");
+    pub const EOA_OWNER: Address = address!("0x3333333333333333333333333333333333333333");
+    pub const CONTRACT_REJECTING_ETH: Address =
+        address!("0x1111111111111111111111111111111111111111");
+    pub const UNKNOWN_ETHFLOW: Address = address!("0x9999999999999999999999999999999999999999");
```
Comment (jmg-duarte) [inline crates/refunder/src/refund_service.rs]: Use repeat_byte

File: crates/refunder/src/refund_service.rs [hunk 6]
```diff
@@ -292,59 +215,696 @@ impl RefundService {
...
+        mock_chain
+            .expect_get_order_status()
+            .withf(|_, order_hash| order_hash.0[31] == 1)
+            .returning(|_, _| Ok(RefundStatus::NotYetRefunded(EOA_OWNER)));
+
+        // Order 2 (uid_suffix=2) was already refunded on-chain, should be filtered out
+        mock_chain
+            .expect_get_order_status()
+            .withf(|_, order_hash| order_hash.0[31] == 2)
+            .returning(|_, _| Ok(RefundStatus::Refunded));
```
Comment (jmg-duarte) [inline crates/refunder/src/refund_service.rs]: Not a fan of this style of setup for the expectations, it's hard to understand where the "action" actually occurs.

Is there no builder pattern or similar?

File: crates/shared/src/lib.rs
Comment (tilacog) [reply crates/shared/src/lib.rs:1]: I gather this is the `Address::random()` issue.

So, I got this error when running `$ cargo check -p refunder --tests`:

```
    Checking shared v0.1.0 (crates/shared)
error[E0599]: no function or associated item named `random` found for struct `alloy::alloy_primitives::Address` in the current scope
   --> crates/shared/src/price_estimation/trade_verifier/mod.rs:371:50
    |
371 |                     verification.from = Address::random();
    |                                                  ^^^^^^ function or associated item not found in `alloy::alloy_primitives::Address`
    |
note: if you're trying to build a new `alloy::alloy_primitives::Address` consider using one of the following associated functions:
      alloy::alloy_primitives::Address::new
      alloy::alloy_primitives::Address::with_last_byte
      alloy::alloy_primitives::Address::repeat_byte
      alloy::alloy_primitives::Address::from_slice
      and 7 others
   --> /$HOME/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/alloy-primitives-1.4.1/src/bits/address.rs:45:1
    |
 45 | / wrap_fixed_bytes!(
 46 | |     // we implement Display with the checksum, so we don't derive it
 47 | |     extra_derives: [],
 48 | |     /// An Ethereum address, 20 bytes in length.
...   |
 88 | |     pub struct Address<20>;
 89 | | );
    | |_^
    = note: this error originates in the macro `wrap_fixed_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

[ same error for crates/shared/src/lib.rs:53:78 ]
```
I just managed to reproduce this from the latest `main` branch (c7fd798ce)

I'm not sure why this doesn't show up in our CI

File: crates/refunder/src/refund_service.rs (hunk 6 above)
Comment (tilacog) [reply crates/refunder/src/refund_service.rs]: Yeah, I agree that is a bit too much.
That's how bare `mockall` works, and those `expect_` + `returning` get all over the place.

I'll try adding some builders and/or helper fns in the follow-up PR.

File: crates/shared/src/lib.rs
Comment (jmg-duarte) [reply crates/shared/src/lib.rs:1]: This doesn't show up on CI due to feature unification. You just need to add `"rand"` to the alloy feature list in the respective crate

Comment (tilacog) [reply crates/shared/src/lib.rs:1]: Thanks. I've addressed this in e3aa8cede5c760d3d6918b483909fba6743b1606

File: crates/refunder/src/refund_service.rs [hunk 7]
```diff
@@ -37,13 +37,29 @@ pub struct RefundService {
...
+    /// Order is invalid (never created, already freed, or owner cannot receive
+    /// ETH).
     Invalid,
 }
 
+impl From<CoWSwapEthFlow::CoWSwapEthFlow::ordersReturn> for RefundStatus {
+    fn from(value: CoWSwapEthFlow::CoWSwapEthFlow::ordersReturn) -> Self {
+        match value.owner {
+            owner if owner == NO_OWNER => Self::Invalid,
+            owner if owner == INVALIDATED_OWNER => Self::Refunded,
```
Comment (jmg-duarte) [inline crates/refunder/src/refund_service.rs]: this should work
```suggestion
            NO_OWNER => Self::Invalid,
            INVALIDATED_OWNER => Self::Refunded,
```

File: crates/e2e/tests/e2e/refunder.rs [hunk 8]
```diff
@@ -1,22 +1,548 @@
...
+/// time). However, tests requiring actual **settlement** must use real-world
+/// timestamps (`Utc::now()`) because autopilot validates against wall-clock
+/// time. See `refunder_skips_settled_orders` for an example.
+///
+/// Note: Both `onchain` and `services` are leaked (via `Box::leak`) to avoid
+/// lifetime issues. This should be safe enough in test code.
+/// Alternatives to leaking inlcude:
+/// - Transmuting, but requires unsafe and introduces some drop complexity
+/// - Borrow services & onchain structs
+/// - Use Rc/Arc
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: To be honest, I prefer the Arc solution, I find that the leak may be a footgun

File: crates/e2e/tests/e2e/refunder.rs [hunk 9]
```diff
@@ -1,22 +1,548 @@
...
+        services.db().clone(),
+        web3,
+        ethflow_contracts,
+        min_validity_duration,
+        min_price_deviation_bps,
+        Box::new(signer),
+        MAX_GAS_PRICE,
+        START_PRIORITY_FEE_TIP,
+    )
+}
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: For now, this one is a bit redundant, you can just call this RefundService in the create_refund_service above

Comment (jmg-duarte) [reply crates/e2e/tests/e2e/refunder.rs]: Disregard, I didn't notice other calls

File: crates/e2e/tests/e2e/refunder.rs [hunk 10]
```diff
@@ -1,22 +1,548 @@
...
+
+    assert!(
+        expect_refund == (status == RefundStatus::Refunded),
+        "Test failed: {description}.\nExpected refund: {expect_refund}, but got status: {status:?}"
+    );
+}
+
+#[rstest]
+// Tests that orders with slippage below, at, or above the min_price_deviation
+// threshold are refunded according to the SQL >= check.
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/refunder.rs:332]: Does the result depend only on a SQL query? If so, can this be tested in a more lightweight setup?  Maybe expanding the database tests would be enough here? And keep only 1-2 cases with a local node involved. My worry is that each test executes in 5-7s, which adds ~40s to the CI job execution. No strong opinion tho. Especially, if the alternative solution will require almost the same time to execute.

File: crates/e2e/tests/e2e/refunder.rs [hunk 11]
```diff
@@ -1,22 +1,548 @@
...
+    tracing::info!("Running refunder threshold test");
+
+    let setup = RefunderTestSetup::new(web3.clone()).await;
+    let ethflow_contract = setup.ethflow_contract();
+
+    let (ethflow_order, _order_id, valid_to) = setup
+        .create_and_index_ethflow_order(slippage.order, validity.order)
+        .await;
+
+    advance_time_past_expiration(&web3, valid_to).await;
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/refunder.rs:293]: Not sure I understand what stops the order from being settled. Could you add a comment in the code?

**Review by squadgazzz — COMMENTED** [review]
Amazing PR 🚀 
Just a few clarification questions before approving it.

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/refunder.rs]: Agree. Why not simply use Arc?

File: crates/e2e/tests/e2e/refunder.rs (hunk 9 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: Agree. That was a leftover from my revert. Will do

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: After giving it a bit more thought, it seems we can't use `Arc/Rc` here because `RefunderTestSetup>` struct can't hold both `Services<'a>`, which in turn borrows `&'a Onchain`. 

However, if we were to modify the `Services` and `Onchain` structs themselves, I think we could use `Arc/Rc`, but that seems like a PR on its own.

In the meantime, I managed to get it working quite easily using [`ouroboros`](https://github.com/someguynamedjosh/ouroboros) for the self-referencing part.

What do you guys think? Should I stick with `ouroboros`, or do we want to refactor `Services`/`Onchain` to use `Arc/Rc` instead?

File: crates/e2e/tests/e2e/refunder.rs (hunk 11 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:293]: Currently, settlement won't happen because of the time distance between the chain (2020) and autopilot (2026, current). 

All tests' orders are expired at creation time by default because they are created based on chain time. The exception being the `refunder_skips_settled_orders` test, which manually creates valid orders.

I'll document this more thoroughly :+1:

File: crates/e2e/tests/e2e/refunder.rs (hunk 10 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:332]: Fair point. 

Considering SQL alone, then I believe yes. I can see order expiration and slippage being tested at the [`postgres_refundable_ethflow_orders` test](https://github.com/cowprotocol/services/blob/b9ff845cb1f4261c80f5e30d00a74f570a97c6f2/crates/database/src/ethflow_orders.rs#L274)

The current test operates one layer above and verifies the full integration, so I dare to say there's a bit more of value being added here. 

Nonetheless, I'm still not sure what's the best way forward here.

Maybe we can remove the boundary test cases, since they are already covered in the database tests?

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (jmg-duarte) [reply crates/e2e/tests/e2e/refunder.rs]: I know this is probably now what you want to read but IMO we should instead (if possible):
1. refactor the test to match the structure of the other ones (that don't require these tricks — read ouroboros, leaks, etc)
2. fix the E2E testing structure in general (if there's space for it) cc'ing @extrawurst for discussion/visibility
  * since it would tackle a big part of the tech debt
  * at the same time we could shift the E2E specialization from Ilya to you since you'd literally be learning and re-writing them

Fixing the E2E doesn't need to be wholesale, it can be similar to the alloy migration, bit by bit, and with E2E there's no risk of adding new tests and infrastructure for said tests and break previous ones; so we could do new proper structures and migrate the E2E tests bit by bit

I suggest this because:
* while the `Box::leak` usage in tests is most likely ok, it's a footgun waiting to blow
* ouroboros would fix this now but we'd add more tech debt to our existing one
* if you're eventually adding traits etc (like in the first iteration of the PR), it's just a matter of time until you hit this again -> adding more leak/ouroboros -> meaning more debt

File: crates/e2e/tests/e2e/refunder.rs (hunk 10 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/refunder.rs:332]: I mean, we don't need to drop all the test cases, we should keep at least 1. But for example, if changing some test case params only depends on the SQL query output, we should test it in the DB tests to avoid spinning up the protocol, etc, just to ensure the query returns a reasonable result. If changing the test case params involves some other logic on the refunder side, then we should keep everything as is for sure.

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: 100% agree.

I'm working on removing the setup struct entirely, so tests will follow the e2e pattern and we won't bump into this issue anymore.

File: crates/e2e/tests/e2e/refunder.rs (hunk 11 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:293]: I hope the last version (bb5233fad92445c7f979b44243fca3880c23a218) is sufficiently documented, please let me know if I should expand/elaborate more.

File: crates/e2e/tests/e2e/refunder.rs (hunk 10 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:332]: I’m on the fence about this. The current build (bb5233fad92445c7f979b44243fca3880c23a218) tests three scenarios for each threshold:

1. Refund should be triggered
2. Refund should not be triggered
3. Edge case, outcome depends on the specific attribute

I'm OK about dropping the edge cases, but ideally we should still test both the triggering and non-triggering paths...
I'm totally with you on the impact this has on our overall test time though, so I'm not sure how to move forward here.

Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:332]: That said, I think we could get the same confidence about triggering behavior from unit tests, which would be way lighter. We'd need some refactoring involving new traits to get there, but once we have them it should be much better. Do you think it's OK to leave these as they are for now and revisit once we have the unit tests in place?

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: I removed the helper/offending struct in the last commit (bb5233fad92445c7f979b44243fca3880c23a218) and replaced it with a builder that just borrows both `Services` and `Onchain`, so there's no more lifetime wrangling. Hopefully that makes more sense now.

File: crates/e2e/tests/e2e/refunder.rs [hunk 12]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
+    /// Select which ethflow contract to use.
+    fn with_ethflow_index(mut self, index: usize) -> Self {
+        self.ethflow_index = index;
+        self
+    }
+
+    /// Creates the order, mines it on-chain, waits for indexing, and optionally
+    /// invalidates.
+    ///
+    /// Returns: (ExtendedEthFlowOrder, OrderUid, valid_to timestamp)
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: This doesn't add information one isn't getting from the function signature

File: crates/e2e/tests/e2e/refunder.rs [hunk 13]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
+/// Threshold tests verify the refunder's SQL eligibility filters (slippage and
+/// validity duration), not settlement behavior. Orders in these tests use
+/// blockchain time for `valid_to`, which combined with Anvil's genesis
+/// timestamp of Jan 1, 2020 causes the autopilot to reject them as expired (it
+/// validates against wall-clock time). This isolation is intentional: the
+/// refunder uses blockchain time internally, so it correctly processes these
+/// "expired" orders.
+///
+/// For tests requiring actual settlement, use `Utc::now()` for `valid_to`.
+/// See `refunder_skips_settled_orders` for an example.
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: I feel that this should be part of the mod documentation as it pertains to how to writing tests, not attached to a single function

File: crates/e2e/tests/e2e/refunder.rs [hunk 14]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
-        side: OrderQuoteSide::Sell {
-            sell_amount: model::quote::SellAmount::AfterFee { value: sell_amount },
-        },
-        ..Default::default()
-    };
-    let quote_response = services.submit_quote(&quote).await.unwrap();
-    let ethflow_order_2 =
-        ExtendedEthFlowOrder::from_quote(&quote_response, valid_to).include_slippage_bps(9999);
+    )
+    .with_slippage_bps(slippage.order) // Explicit: testing this SQL filter
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: I'm confused as to what this comment means, you're testing this filter?

File: crates/e2e/tests/e2e/refunder.rs [hunk 15]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
+        START_PRIORITY_FEE_TIP,
+    );
+
+    // Verify order is not yet invalidated
+    assert_ne!(
+        ethflow_order
+            .status(onchain.contracts(), ethflow_contract)
+            .await,
+        RefundStatus::Refunded
+    );
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs]: The docstring kind of mismatches the code, it's checking if the order wasn't refunded yet, meaning it can actually be invalid.

If you're expecting the order to not be invalid, you should check for that instead (?)

File: crates/e2e/tests/e2e/refunder.rs [hunk 16]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
+
+    // The order should already be invalidated on-chain before the refunder runs
+    assert_eq!(
+        ethflow_order
+            .status(onchain.contracts(), ethflow_contract)
+            .await,
+        RefundStatus::Refunded,
+        "Order should already be invalidated by user"
+    );
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs:439]: If the order is invalidated on chain, why are you checking against refunded?

File: crates/e2e/tests/e2e/refunder.rs [hunk 17]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
+        .await
+        .unwrap();
+
+    // The order should still be invalidated (status unchanged)
+    assert_eq!(
+        ethflow_order
+            .status(onchain.contracts(), ethflow_contract)
+            .await,
+        RefundStatus::Refunded
+    );
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs:453]: Ditto

File: crates/e2e/tests/e2e/refunder.rs [hunk 18]
```diff
@@ -48,123 +90,595 @@ async fn refunder_tx(web3: Web3) {
...
-    web3.alloy
-        .evm_mine(None)
+    // The order should still NOT be invalidated (refunder skipped it)
+    assert_ne!(
+        ethflow_order
+            .status(onchain.contracts(), ethflow_contract)
+            .await,
+        RefundStatus::Refunded,
+        "Refunder should not have invalidated the already-settled order"
+    );
```
Comment (jmg-duarte) [inline crates/e2e/tests/e2e/refunder.rs:552]: There doesn't seem to be a proper status for settled orders. Should this actually be invalid in the face of the refunder since it can't be refunded?

File: crates/e2e/tests/e2e/refunder.rs (hunk 10 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/refunder.rs:332]: Let's leave it as is then. If this requires additional work, I think we can live with an extra 30s of CI running.

File: crates/e2e/tests/e2e/refunder.rs (hunk 16 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:439]: The `RefundStatus::Refunded` variant is a bit confusing because it's semantic overloaded. It represents both:
- Orders refunded by the refunder service, and
- Orders invalidated/cancelled by the user

So, `Refunded` is the variant we have for a refund in a terminal state.

https://github.com/cowprotocol/services/blob/bb5233fad92445c7f979b44243fca3880c23a218/crates/refunder/src/refund_service.rs#L40-L61

**Note:** The `impl From` above was originally at [`crates/e2e/tests/e2e/ethflow.rs#L796-L804` in the `main` branch,](https://github.com/cowprotocol/services/blob/b9ff845cb1f4261c80f5e30d00a74f570a97c6f2/crates/e2e/tests/e2e/ethflow.rs#L789-L804) before I consolidated the `EthFlowOrderOnchainStatus` enum w/ the `RefundStatus` I took from test code. I tried to not loose anything in translation, but now I'm having second thoughts.

Not sure how to go from here. Should I add more comments to explain this further? I'd rather avoid expanding/specializing the variants in this PR.

Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:439]: (Not sure I'm missing something here too)

File: crates/e2e/tests/e2e/refunder.rs (hunk 17 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:453]: I'm covering this in my [previous reply](https://github.com/cowprotocol/services/pull/4029#discussion_r2683419059).

File: crates/e2e/tests/e2e/refunder.rs (hunk 18 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs:552]: I'm covering this in my [previous reply](https://github.com/cowprotocol/services/pull/4029#discussion_r2683419059).

File: crates/e2e/tests/e2e/refunder.rs (hunk 15 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: Bad wording on my end, and also poor logic. 

What we should be testing here can be reworded to _"checking if the order is still eligible for refund (not yet reimbursed)"_ or "check if order has not been refunded yet".

Fixed the comment in a760a8927

File: crates/e2e/tests/e2e/refunder.rs (hunk 14 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: Possibly a leftover comment from a previous version, my bad.
Removed in 5c493b2ac

File: crates/e2e/tests/e2e/refunder.rs (hunk 13 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: Addressed in cfbf74f4c

File: crates/e2e/tests/e2e/refunder.rs (hunk 12 above)
Comment (tilacog) [reply crates/e2e/tests/e2e/refunder.rs]: removed in e7d939221

File: crates/e2e/tests/e2e/refunder.rs (hunk 8 above)
Comment (jmg-duarte) [reply crates/e2e/tests/e2e/refunder.rs]: It very much does, thanks

File: crates/e2e/tests/e2e/refunder.rs (hunk 16 above)
Comment (jmg-duarte) [reply crates/e2e/tests/e2e/refunder.rs:439]: This definitely needs improvement then, not in this PR though
